flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarListHandler.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.webmonitor.handlers;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import org.apache.flink.client.program.PackagedProgram;
+import org.apache.flink.client.program.ProgramInvocationException;
+import org.apache.flink.runtime.instance.ActorGateway;
+import org.apache.flink.runtime.webmonitor.RuntimeMonitorHandler;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Map;
+import java.util.UUID;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+public class JarListHandler implements RequestHandler, RequestHandler.JsonResponse {
+
+	private final File jarDir;
+
+	public  JarListHandler(File jarDirectory) {
+		jarDir = jarDirectory;
+	}
+
+	@Override
+	public String handleRequest(Map<String, String> params, ActorGateway jobManager) throws Exception {
+		try {
+			StringWriter writer = new StringWriter();
+			JsonGenerator gen = JsonFactory.jacksonFactory.createJsonGenerator(writer);
+			gen.writeStartObject();
+			gen.writeStringField("address", params.get(RuntimeMonitorHandler.WEB_MONITOR_ADDRESS_KEY));
+			if (jarDir != null) {
```

Comment:
That's exactly why I offered to make it more explicit. :')
On Nov 17, 2015 11:26 PM, "Robert Metzger" <notifications@github.com> wrote:

> In
> flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarListHandler.java
> <https://github.com/apache/flink/pull/1338#discussion_r45094049>:
>
> > +public class JarListHandler implements RequestHandler, RequestHandler.JsonResponse {
> > +
> > +	private final File jarDir;
> > +
> > +	public  JarListHandler(File jarDirectory) {
> > +		jarDir = jarDirectory;
> > +	}
> > +
> > +	@Override
> > +	public String handleRequest(Map<String, String> params, ActorGateway jobManager) throws Exception {
> > +		try {
> > +			StringWriter writer = new StringWriter();
> > +			JsonGenerator gen = JsonFactory.jacksonFactory.createJsonGenerator(writer);
> > +			gen.writeStartObject();
> > +			gen.writeStringField("address", params.get(RuntimeMonitorHandler.WEB_MONITOR_ADDRESS_KEY));
> > +			if (jarDir != null) {
>
> I don't doubt that its working correctly now. The issue is that its really
> not obvious what this check is for. Somebody else changing your code might
> easily break it!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/flink/pull/1338/files#r45094049>.
>